useragent/useragent.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package useragent has functions for setting a user agent with helpful information
+package useragent
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/sonatype-nexus-community/nancy/buildversion"
+)
+
+var GOOS = runtime.GOOS
+var GOARCH = runtime.GOARCH
+
+func GetUserAgent() (useragent string) {
+	useragent = fmt.Sprintf("nancy-client/%s", buildversion.BuildVersion)
+	if checkForCIEnvironment() {
+		callerInfo := getCallerInfo()
+		if callerInfo == "" {
+			useragent = checkCIEnvironments(useragent)
+			return
+		}
+		useragent = useragent + fmt.Sprintf(" (%s; %s %s)", callerInfo, GOOS, GOARCH)
+		return
+	}
+	useragent = useragent + fmt.Sprintf(" (%s; %s %s)", "non ci usage", GOOS, GOARCH)
```

Comment:
Why not just put the useragent inside the Sprintf instead of combining strings 2 ways?

Reply:
Because: reasons (no reason)

Reply:
In: 9458664